ppoprf/src/ppoprf.rs

```diff
@@ -100,6 +263,17 @@ impl ProofDLEQ {
     RistrettoScalar::from_bytes_mod_order_wide(&out)
   }
 
+  // I2OSP2(x): Converts a non-negative integer x into a byte
+  // array of length 2 as described in [RFC8017]. Note that
+  // this function returns a byte array in big-endian byte order.
+  fn i2osp2(x: usize) -> [u8; 2] {
+    let x_u16: u16 = x.try_into().expect("integer too large");
+    let y = x_u16.to_be_bytes();
+    let mut z = [0u8; 2];
+    z.clone_from_slice(&y[y.len() - 2..]);
+    z
```

Comment:
You can just return `y` directly here. `u16::to_be_bytes()` is already a `[u8; 2]` so it's not necessary to copy out the last two bytes.